fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java

```diff
@@ -129,20 +132,26 @@ public Optional<Snapshot> getSnapshot() {
         }
     }
 
-    public long getSnapshotId() {
-        return snapshotId;
-    }
-
-    public void setSnapshotId(long snapshotId) {
-        this.snapshotId = snapshotId;
+    public long getCachedSnapshotId() {
+        return cachedSnapshotId;
     }
 
     public List<String> getCachedPartitionNames() {
-        return cachedPartitionNames;
+        readLock();
+        try {
+            return cachedPartitionNames;
+        } finally {
+            readUnlock();
+        }
     }
 
     public void setCachedPartitionNames(List<String> cachedPartitionNames) {
-        this.cachedPartitionNames = cachedPartitionNames;
+        writeLock();
+        try {
+            this.cachedPartitionNames = cachedPartitionNames;
```

Comment:
update the `cachedSnapshotId` and `cachedPartitionNames` in the same critical section ?

Reply:
noly cachedPartitionNames,  cachedSnapshotId would be assigned at new IcebergTable